### ISSUE TITLE
Improved default enum values for typescript

### ISSUE THREAD

**Issue by jedevc** [dagger/dagger#8152]
Spun out of https://github.com/dagger/dagger/pull/8115#discussion_r1716905912

Currently, a default enum value might look something like:

```typescript
constructor(status: Status = "INACTIVE") {
```

However, ideally we would express it as:

```typescript
constructor(status: Status = Status.Inactive) {
```

Using the actual value of the enum defined.

However, as mentioned by @TomChv, this is a bit tricksy:

>> I suspect this is because of [main/sdk/typescript/introspector/scanner/abtractions/argument.ts#L162](https://github.com/dagger/dagger/blob/main/sdk/typescript/introspector/scanner/abtractions/argument.ts?rgh-link-date=2024-08-14T11%3A20%3A04Z#L162) - I think getText isn't correct here, I guess that's setting the default directly to Status.Inactive, which can't be JSON unmarshaled in the core.
>
> That's right, there's no such evalutation by reference in the TS SDK, I'm not sure how to fix it, I'll need to add a new capabilities to the SDK.

**Comment by TomChv** [comment]
Yeah to do such thing, I need to resolve value by reference, I'm currently working on a proof of concept for that though.

**Comment by TomChv** [comment]
Related to https://github.com/dagger/dagger/issues/8676

**Comment by TomChv** [comment]
I'm closing this issue since it's fixed by #8824 :)
